ui/admin/app/components/form/target/details/index.js

```diff
@@ -57,31 +54,30 @@ export default class FormTargetComponent extends Component {
    * @type {boolean}
    */
   get showUpdateWorkerFilterButton() {
-    return (
-      !this.args.model.isNew &&
-      this.args.model.worker_filter &&
-      !this.migrateWorkerFilter
-    );
+    return this.args.model.worker_filter?.length;
+  }
+
+  get showEgressFilter() {
+    return !this.args.model.worker_filter?.length && !this.args.model.isNew;
   }
 
   //actions
   @action
   toggleEgressWorkerFilter() {
     this.egressWorkerFilterEnabled = !this.egressWorkerFilterEnabled;
   }
+
   // =actions
   @action
   migrateWorkerFilters() {
-    this.migrateWorkerFilter = true;
     this.egressWorkerFilterEnabled = true;
-    // When update is clicked, copy worker filter value into egress filter.
+    // When update is clicked, copy worker filter value into egress filter and clear the worker_filter
     this.args.model.egress_worker_filter = this.args.model.worker_filter;
     this.args.model.worker_filter = '';
   }
 
   @action
   @loading
-  @notifyError(({ message }) => message)
```

Comment:
Won't we stop notifying errors now if removingHostSources fail? I think this is still needed.